authz/providers/azure/options.go

```diff
@@ -0,0 +1,125 @@
+)
+
+type Options struct {
+	AuthzMode                    string
+	ResourceId                   string
+	AKSAuthzURL                  string
+	ARMCallLimit                 int
+	SkipAuthzCheck               []string
+	AuthzResolveGroupMemberships bool
+	SkipAuthzForNonAADUsers      bool
+}
+
+func NewOptions() Options {
+	return Options{}
+}
+
+func (o *Options) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&o.AuthzMode, "azure.authz-mode", "", "authz mode to call RBAC api, valid value is either aks or arc")
+	fs.StringVar(&o.ResourceId, "azure.resource-id", "", "azure cluster resource id (//subscription/<subName>/resourcegroups/<RGname>/providers/Microsoft.ContainerService/managedClusters/<clustername> for AKS or //subscription/<subName>/resourcegroups/<RGname>/providers/Microsoft.Kubernetes/connectedClusters/<clustername> for arc) to be used as scope for RBAC check")
+	fs.StringVar(&o.AKSAuthzURL, "azure.aks-authz-url", "", "url to call for AKS Authz flow")
+	fs.IntVar(&o.ARMCallLimit, "azure.arm-call-limit", defaultArmCallLimit, "No of calls before which webhook switch to new ARM instance to avoid throttling")
+	fs.StringSliceVar(&o.SkipAuthzCheck, "azure.skip-authz-check", []string{""}, "name of usernames/email for which authz check will be skipped")
+	fs.BoolVar(&o.AuthzResolveGroupMemberships, "azure.authz-resolve-group-memberships", o.AuthzResolveGroupMemberships, "set to true to resolve group membership by authorizer. Setting to false will use group list from subjectaccessreview request")
+	fs.BoolVar(&o.SkipAuthzForNonAADUsers, "azure.skip-authz-for-non-aad-users", o.SkipAuthzForNonAADUsers, "skip authz for non AAD users")
+}
+
+func (o *Options) Validate(azure azure.Options) []error {
+	var errs []error
+	o.AuthzMode = strings.ToLower(o.AuthzMode)
+	switch o.AuthzMode {
+	case AKSAuthzMode:
+	case ARCAuthzMode:
+	case "":
+	default:
+		errs = append(errs, errors.New("invalid azure.authz-mode. valid value is either aks or arc"))
+	}
+
+	if o.AuthzMode != "" && o.ResourceId == "" {
+		errs = append(errs, errors.New("azure.resource-id must be non-empty for authorization"))
+	}
+
+	if o.AuthzMode == AKSAuthzMode && o.AKSAuthzURL == "" {
+		errs = append(errs, errors.New("azure.aks-authz-url must be non-empty"))
+	}
+
+	if o.AuthzMode != AKSAuthzMode && o.AKSAuthzURL != "" {
+		errs = append(errs, errors.New("azure.aks-authz-url must be set only with AKS authz mode"))
+	}
+
+	if o.AuthzMode == ARCAuthzMode {
+		if azure.ClientSecret == "" {
+			errs = append(errs, errors.New("azure.client-secret must be non-empty"))
+		}
+		if azure.ClientID == "" {
+			errs = append(errs, errors.New("azure.client-id must be non-empty"))
+		}
+	}
+
+	if o.ARMCallLimit > maxPermissibleArmCallLimit {
+		errs = append(errs, errors.New("azure.arm-call-limit must not be more than 4000"))
+	}
+
+	return errs
+}
+
+func (o Options) Apply(d *apps.Deployment) (extraObjs []runtime.Object, err error) {
+	container := d.Spec.Template.Spec.Containers[0]
```

Comment:
Why not just have the `args` variable? I don't see the container variable is used here.

`args := d.Spec.Template.Spec.Containers[0].Args`